### PR TITLE
homework_2 and fix hw_1_3 with append in loop

### PR DESCRIPTION
**Pull request by ColeMaddison**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
+{
```
Comment (mmykhailenko) [inline .vscode/launch.json:1]: add this file to .gitignor

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (mmykhailenko) [inline .vscode/settings.json:1]: add this file to .gitignor

File: python_2/hw_2_1.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+#1.2) сгруппировать данные по значению ключа 'city'
+
+from itertools import groupby
+
+def list_group_city(data):
+    list_groupped = []
+    data.sort(key=lambda item: item['city'])
+    for key, group in groupby(data, lambda x: x['city']):
+        list_groupped.append({key: list({k: i[k] for k in i.keys() if not k == 'city'} for i in group)})
```
Comment (mmykhailenko) [inline python_2/hw_2_1.py]: list({k: i[k] for k in i.keys() if not k == 'city'} - this should be moved to variable

File: python_2/hw_2_5.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# 5) Есть строка со словами и числами, разделенными пробелами (один пробел между словами и/или числами).
+# Слова состоят только из букв. Вам нужно проверить есть ли в исходной строке три слова подряд.
+# Для примера, в строке "hello 1 one two three 15 world" есть три слова подряд.
+
+import re
+
+string_1 = 'hello 1 one two three 15 world'
```
Comment (mmykhailenko) [inline python_2/hw_2_5.py:7]: will not work -  'hello 1 one two three four 15 world'
will not work -  'hello 1 one two three four five 15 world' ...

File: python_2/hw_2_3.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# 3) Дано целое число. Необходимо подсчитать произведение всех цифр в этом числе, за исключением нулей.
+# Например:
+# Дано число 123405. Результат будет: 1*2*3*4*5=120.
+
+from functools import reduce
+
+number = 123405
+
+def count_prod(number):
+    return reduce((lambda x, y: (int(x) * int(y))), list(filter(lambda i: i!='0', str(number))))
```
Comment (mmykhailenko) [inline python_2/hw_2_3.py]: i!='0' - this is bad construction, so code should be rewritten
Example:
```
res = 1
for d in str(number):
    res *= int(d) if int(d) else 1
return res
```

**Review by mmykhailenko — APPROVED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (ColeMaddison) [reply .vscode/launch.json:1]: done

File: .vscode/settings.json (hunk 2 above)
Comment (ColeMaddison) [reply .vscode/settings.json:1]: done

File: python_2/hw_2_1.py (hunk 3 above)
Comment (ColeMaddison) [reply python_2/hw_2_1.py]: done

File: python_2/hw_2_5.py (hunk 4 above)
Comment (ColeMaddison) [reply python_2/hw_2_5.py:7]: did not understand the comment, here is the out put for three cases:

# input
string_1 = 'hello 1 one two three 15 world'
string_2 = '1 23'
string_3 = 'hello 1 one two three four five 15 world'

# function call
print(find_consecutive_words(string_1))
print(find_consecutive_words(string_2))
print(find_consecutive_words(string_3))

# output
True
False
True

File: python_2/hw_2_3.py (hunk 5 above)
Comment (ColeMaddison) [reply python_2/hw_2_3.py]: i agree, that solution is much more simple
should use less functional programming
rewrote

**Pull request merged**